### ISSUE TITLE
File upload with craft 2.5 fails

### ISSUE THREAD

**Issue by timbertens** [am-impact/amforms#29]
Since the upgrade to craft 2.5 (using the latest AM Forms release 1.2.10, but also had it with 1.2.8) uploading a file using a A&M Forms does not work anymore.  I'm not 100% sure it is actually a A&M Forms issue, could be a general Craft 2.5 issue.  This is the exact error:

PHP warning
array_merge(): Argument #1 is not an array
/Applications/MAMP/htdocs/denoordboom.local/craft/app/fieldtypes/AssetsFieldType.php(266)

and this is the code that fails:

```
254             }
255 
256             foreach ($uploadedFiles as $file)
257             {
258                 $tempPath = AssetsHelper::getTempFilePath($file['filename']);
259                 move_uploaded_file($file['location'], $tempPath);
260                 $response = craft()->assets->insertFileByLocalPath($tempPath, $file['filename'], $targetFolderId);
261                 $fileIds[] = $response->getDataItem('fileId');
262                 IOHelper::deleteFile($tempPath, true);
263             }
264         }
265 
266         $fileIds = array_merge($value, $fileIds);
267 
268         // Make it look like the actual POST data contained these file IDs as well,
269         // so they make it into entry draft/version data
270         $this->element->setRawPostContent($this->model->handle, $fileIds);
271 
272         return $fileIds;
273     }
274 
275     /**
276      * @inheritDoc IFieldType::onAfterElementSave()
277      *
278      * @return null
```

**Comment by bvaillant** [comment]
It look like its a Craft issues.

If you want to resolve this issues quick, while waiting on a update from Craft.

Add these line after line 241 of /craft/app/fieldtypes/AssetsFieldType.php

```
if (empty($value)) {
    $value = array();
}
```

**Comment by timbertens** [comment]
I've reported it to pixel and tonic and it is already fixed in the latest release 2.5.2752
thanks for your support, all is fine now!

**Comment by oadslug** [comment]
This also seems to be happening with Craft CMS 2.6.2776.

If 'Limit the number of selectable assets' is set to 1, it returns this error. 
```
array_merge(): Argument #1 is not an array
```

If 'Limit the number of selectable assets' is set to more than 1, it returns this error. 
```
CDbCommand failed to execute the SQL statement: SQLSTATE[23000]: Integrity constraint violation: 1452 Cannot add or update a child row: a foreign key constraint fails (`craft_oxbow`.`craft_relations`, CONSTRAINT `craft_relations_targetId_fk` FOREIGN KEY (`targetId`) REFERENCES `craft_elements` (`id`) ON DELETE CASCADE). The SQL statement executed was: INSERT INTO `craft_relations` (`fieldId`, `sourceId`, `sourceLocale`, `targetId`, `sortOrder`, `dateCreated`, `dateUpdated`, `uid`) VALUES (:row0_col0, :row0_col1, NULL, :row0_col3, :row0_col4, :row0_col5, :row0_col6, :row0_col7). Bound with :row0_col0='107', :row0_col1='4753', :row0_col3='alison-wright-photo.jpg', :row0_col4=1, :row0_col5='2016-03-20 23:00:19', :row0_col6='2016-03-20 23:00:19', :row0_col7='2c28ebdf-496b-4217-80c2-c8cddcb6d6bf'
```
